### PR TITLE
Refactor hba1c example ehrQL snippet

### PR DESCRIPTION
**Pull request by madwort**
* more re-use of fragments
* demonstrate sorting on two keys
* after some [discussion in slack](https://bennettoxford.slack.com/archives/C069YDR4NCA/p1731501153163389)

### CODE REVIEW

File: docs/how-to/examples.md [hunk 1]
```diff
@@ -519,18 +519,39 @@ from ehrql.tables.core import clinical_events, patients
...
         clinical_events.date.is_on_or_after("2022-07-01")
-).numeric_value.maximum_for_patient()
+)
+
+earliest_min_hba1c_event = hba1c_events.sort_by(
+        clinical_events.numeric_value, clinical_events.date
+).first_for_patient()
+
+earliest_max_hba1c_event = hba1c_events.sort_by(
+        -clinical_events.numeric_value, clinical_events.date
```
Comment (evansd) [inline docs/how-to/examples.md:534]: I reckon this is worth a comment, something like:
```suggestion
        # Note the leading minus sign to sort numeric_value in reverse order
        -clinical_events.numeric_value, clinical_events.date
```

Particularly on the assumption that this example will get copied around.

File: docs/how-to/examples.md [hunk 2]
```diff
@@ -519,18 +519,39 @@ from ehrql.tables.core import clinical_events, patients
...
+).first_for_patient()
+
+latest_min_hba1c_event = hba1c_events.sort_by(
+        -clinical_events.numeric_value, clinical_events.date
```
Comment (evansd) [inline docs/how-to/examples.md:538]: Same for this.

**Review by evansd — APPROVED** [review]
Looks good. Left a comment about a comment.

I'd be tempted to squash merge this for the sake of a cleaner history as I don't the individual commits add much.

File: docs/how-to/examples.md (hunk 2 above)
Comment (madwort) [reply docs/how-to/examples.md:538]:
```suggestion
        # Note the leading minus sign to sort numeric_value in reverse order
        -clinical_events.numeric_value, clinical_events.date
```

**Pull request merged**